pandas/core/generic.py

```diff
@@ -2494,80 +2494,91 @@ def to_xarray(self):
-            C        (index) float64 4.0 5.0 6.0
-
-        >>> df = pd.DataFrame({'A' : [1, 1, 2],
-                               'B' : ['foo', 'bar', 'foo'],
-                               'C' : np.arange(4.,7)}
-                             ).set_index(['B','A'])
-        >>> df
-                 C
-        B   A
-        foo 1  4.0
-        bar 1  5.0
-        foo 2  6.0
-
-        >>> df.to_xarray()
+            name       (index) object 'falcon' 'parrot' 'lion' 'monkey'
+            class      (index) object 'bird' 'bird' 'mammal' 'mammal'
+            max_speed  (index) float64 389.0 24.0 80.5 nan
+            num_legs   (index) int64 2 2 4 4
+
+        >>> df_multiindex = df.set_index(['class', 'name'])
+        >>> df_multiindex
+                       max_speed  num_legs
+        class  name
+        bird   falcon      389.0         2
+               parrot       24.0         2
+        mammal lion         80.5         4
+               monkey        NaN         4
+
+        >>> df_multiindex.to_xarray()
         <xarray.Dataset>
-        Dimensions:  (A: 2, B: 2)
+        Dimensions:    (class: 2, name: 4)
         Coordinates:
-          * B        (B) object 'bar' 'foo'
-          * A        (A) int64 1 2
+          * class      (class) object 'bird' 'mammal'
+          * name       (name) object 'falcon' 'lion' 'monkey' 'parrot'
         Data variables:
-            C        (B, A) float64 5.0 nan 4.0 6.0
-
-        >>> p = pd.Panel(np.arange(24).reshape(4,3,2),
-                         items=list('ABCD'),
-                         major_axis=pd.date_range('20130101', periods=3),
-                         minor_axis=['first', 'second'])
-        >>> p
-        <class 'pandas.core.panel.Panel'>
-        Dimensions: 4 (items) x 3 (major_axis) x 2 (minor_axis)
-        Items axis: A to D
-        Major_axis axis: 2013-01-01 00:00:00 to 2013-01-03 00:00:00
-        Minor_axis axis: first to second
-
-        >>> p.to_xarray()
-        <xarray.DataArray (items: 4, major_axis: 3, minor_axis: 2)>
-        array([[[ 0,  1],
-                [ 2,  3],
-                [ 4,  5]],
-               [[ 6,  7],
-                [ 8,  9],
-                [10, 11]],
-               [[12, 13],
-                [14, 15],
-                [16, 17]],
-               [[18, 19],
-                [20, 21],
-                [22, 23]]])
+            max_speed  (class, name) float64 389.0 nan nan 24.0 nan 80.5 nan nan
+            num_legs   (class, name) float64 2.0 nan nan 2.0 nan 4.0 4.0 nan
+
+        >>> index = pd.MultiIndex(levels=[[pd.to_datetime("2018-01-01"),
+        ... pd.to_datetime("2015-05-23"), pd.to_datetime("2015-06-06"),
+        ... pd.to_datetime("2011-02-13"), pd.to_datetime("2014-07-06")],
+        ...    ['one', 'two']],
+        ...   labels=[[0, 0, 1, 1, 2, 2, 3, 3], [0, 1, 0, 1, 0, 1, 0, 1]],
+        ...   names=['first', 'second'])
+
+        >>> s = pd.Series(np.arange(8), index=index)
+        >>> s
```

Comment:
I find this too complicated for what we need to show. To have a Series with a multiindex with a datetime level, we can have something like:

```
import pandas as pd

df = pd.DataFrame({'date': pd.to_datetime(['2018-01-01', '2018-01-01', '2018-01-02', '2018-01-02']),
                   'animal': ['falcon', 'parrot', 'falcon', 'parrot'],
                   'speed': [350, 18, 361, 15]}).set_index(['date', 'animal'])
df['speed']
```

I haven't used much xarray myself, and not sure what makes sense to show here. May be:
- Series.to_xarray()
- DataFrame.to_xarray()
- DataFrame(with multiindex including datetime).to_xarray()

If that makes sense, I think with the first example, we can have `df.to_xarray()` and `df['max_speed'].to_xarray()`, and then a example like the one I wrote.

@jreback does this make sense?

Sorry for requesting the changes @Moisan, but my I find like the current version gives the idea that we're trying to show something more complex than what we are actually showing.

Reply:
No problem, I'm happy to make the examples more relevant :).